models/users.py

```diff
@@ -41,6 +43,20 @@ def get_id(self) -> UserId:
         """
         return self.id
 
+    def set_password(self, value: str):
+        """
+        Sets a hashed password for user using bcrypt
+        """
+        b_value = bytes(value, 'utf-8')
+        self.password = bcrypt.hashpw(b_value, bcrypt.gensalt())
+
+    def check_password(self, value: str) -> bool:
+        """
+        Checks if value matches the user's password and returns a boolean
+        """
+        b_value = bytes(value, 'utf-8')
+        return bcrypt.checkpw(b_value, self.hashed_password)
```

Comment:
The methods are actually perfect and work well, but (nitpicking here) the variable names don't actually show the intent of what is happening. Some small name refactorings here would go a long way for code clarity.